src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -408,5 +381,46 @@ public ValueTask<(Solution solution, bool updated)> TryUpdateWorkspaceCurrentSol
                 return tuple.solution;
             }
         }
+
+        private sealed class LazySolution : IAsyncDisposable, IDisposable
+        {
+            private readonly RemoteWorkspace _remoteWorkspace;
+            private readonly Checksum _solutionChecksum;
+            private readonly CancellationTokenSource _cancellationTokenSource = new();
+            private readonly Task<Solution> _task;
+
+            public LazySolution(RemoteWorkspace remoteWorkspace, Checksum solutionChecksum, Func<CancellationToken, Task<Solution>> getSolutionAsync)
+            {
+                _remoteWorkspace = remoteWorkspace;
+                _solutionChecksum = solutionChecksum;
+                _task = getSolutionAsync(_cancellationTokenSource.Token);
```

Comment:
mixed feelings about having to pass the getSolutionAsync method into this.  Could just have hte caller create the task and pass it in.

Reply:
that causes a weird ownership transfer for the disposable resource (CancellationTokenSource) that I really wanted to avoid

Reply:
Oh you're right. I missed which CT was being used here. Ignore this.